Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/extrude.h

```diff
@@ -0,0 +1,321 @@
+                  const NamedParameters1& np_in,
+                  const NamedParameters2& np_out)
+{
+  typedef typename boost::graph_traits<InputMesh>::vertex_descriptor input_vertex_descriptor;
+  typedef typename boost::graph_traits<InputMesh>::halfedge_descriptor input_halfedge_descriptor;
+
+  typedef typename boost::graph_traits<OutputMesh>::vertex_descriptor   output_vertex_descriptor;
+  typedef typename boost::graph_traits<OutputMesh>::halfedge_descriptor output_halfedge_descriptor;
+
+  CGAL_assertion(!CGAL::is_closed(input));
+  typedef typename GetVertexPointMap < OutputMesh, NamedParameters2>::type VPMap;
+  typedef typename GetVertexPointMap < InputMesh, NamedParameters1>::const_type IVPMap;
+
+  VPMap output_vpm = choose_param(get_param(np_out, internal_np::vertex_point),
+                                  get_property_map(vertex_point, output));
+  IVPMap input_vpm = choose_param(get_param(np_in, internal_np::vertex_point),
+                                  get_const_property_map(vertex_point, input));
+
+  std::vector<std::pair<input_vertex_descriptor, output_vertex_descriptor> > bottom_v2v;
+  std::vector<std::pair<input_halfedge_descriptor, output_halfedge_descriptor> > bottom_h2h;
+  copy_face_graph(input, output, std::back_inserter(bottom_v2v),
+                  std::back_inserter(bottom_h2h), Emptyset_iterator(),
+                  input_vpm, output_vpm);
+
+  // create the offset for the other side
+  for(std::size_t i = 0; i< bottom_v2v.size(); ++i)
+  {
+    bot(bottom_v2v[i].first, bottom_v2v[i].second);
+  }
+  CGAL::Polygon_mesh_processing::reverse_face_orientations(output);
+
+  // collect border halfedges for the creation of the triangle strip
+  std::vector<std::pair<input_vertex_descriptor, output_vertex_descriptor> > top_v2v;
+  std::vector<std::pair<input_halfedge_descriptor, output_halfedge_descriptor> > top_h2h;
+  copy_face_graph(input, output, std::inserter(top_v2v, top_v2v.end()),
+                  std::inserter(top_h2h, top_h2h.end()), Emptyset_iterator(),
+                  input_vpm, output_vpm);
+  for(std::size_t i = 0; i< top_v2v.size(); ++i)
+  {
+    top(top_v2v[i].first, top_v2v[i].second);
+  }
+  std::vector<output_halfedge_descriptor> border_hedges;
+  std::vector<output_halfedge_descriptor> offset_border_hedges;
+  for(std::size_t i = 0; i< top_h2h.size(); ++i)
+  {
+    input_halfedge_descriptor h = top_h2h[i].first;
+    if( CGAL::is_border(h, input) )
+    {
+      border_hedges.push_back(top_h2h[i].second);
+      offset_border_hedges.push_back(bottom_h2h[i].second);
+      CGAL_assertion(is_border(border_hedges.back(), output));
+      CGAL_assertion(is_border(offset_border_hedges.back(), output));
+    }
+  }
+  // now create a triangle strip
+  extrude_impl::create_strip(border_hedges, offset_border_hedges, output);
+}
+
+
+/**
+ * \ingroup PMP_meshing_grp
+ * fills `output` with a close mesh bounding the volume swept by `input` when translating its
```

Comment:
"close mesh" should be "closed surface mesh"
(I think here "close" sounds like "next to")